### PR TITLE
feat: Add internal reviews on top of the technical review

### PR DESCRIPTION
**Pull request by martin-trajanovski**
## Description

This PR aims to add the new feature called `Internal Reviews`. Those are reviews that are used only internally as a help for the technical review on a proposal. We can have as many internal reviews as we want per technical review on a proposal. There is even a new role introduced called `INTERNAL_REVIEWER` which is used to filter people assigned to the internal review.

## Motivation and Context

Previously (in the old way and older software) this was more or less done by phone as a process of consulting more people/reviewers about their opinion about the technical review of a proposal. This is much better way because we keep track of it in the system. The reviews are not used on a proposal later as part of it's pdf or anything similar.

## How Has This Been Tested

- manual tests

## Fixes

https://jira.esss.lu.se/browse/SWAP-3244

https://github.com/UserOfficeProject/user-office-core/assets/6333063/bbc874d0-9385-4b85-812e-f5179aff9042

## Tests included/Docs Updated?

- [ ] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: apps/user-office-backend/src/datasources/postgres/InternalReviewDataSource.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import { DeleteInternalReviewInput } from '../../resolvers/mutations/internalReview/DeleteInternalReviewMutation';
+import { UpdateInternalReviewInput } from '../../resolvers/mutations/internalReview/UpdateInternalReviewMutation';
+import { InternalReviewsFilter } from '../../resolvers/queries/InternalReviewsQuery';
+import { InternalReviewDataSource } from '../InternalReviewDataSource';
+import { ReviewDataSource } from '../ReviewDataSource';
+import database from './database';
+import { InternalReviewRecord } from './records';
+
+@injectable()
+export default class PostgresInstrumentDataSource
```
[the remark this answers is not in the document]
Comment (martin-trajanovski) [reply apps/user-office-backend/src/datasources/postgres/InternalReviewDataSource.ts:17]: You are right. I need to rename this.

File: apps/user-office-backend/src/queries/InternalReviewQueries.spec.ts [hunk 2]
```diff
@@ -1,50 +1,64 @@
...
+    null
+  );
+});
+
+test('A not logged in user can not get all internal reviews', () => {
+  return expect(internalReviewQueries.getAll(null)).resolves.toBe(null);
 });
 
-test('A instrument scientists can not get other scientists calls', () => {
+test('Instrument scientist can get internal reviews on their instrument', () => {
```
Comment (Scott-James-Hurley) [inline apps/user-office-backend/src/queries/InternalReviewQueries.spec.ts:50]: Would it be an idea to make `dummyInternalReview` and `anotherDummyInternalReview` test data for testing that two instrument scientists in charge of different instruments don't see internal reviews for each other's instruments (if I'm understanding correctly how it works)? You could then test that the User Officer can see both?

**Review by Scott-James-Hurley — COMMENTED** [review]
Thanks very much for addressing my feedback.

**Review by Scott-James-Hurley — APPROVED** [review]

**Review by enthusiastio — APPROVED** [review]
Just a small comment :+1:

File: apps/user-office-backend/src/mutations/InternalReviewMutations.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import { UpdateInternalReviewInput } from '../resolvers/mutations/internalReview/UpdateInternalReviewMutation';
+
+@injectable()
+export default class InternalReviewMutations {
+  constructor(
+    @inject(Tokens.InternalReviewDataSource)
+    private internalReviewDataSource: InternalReviewDataSource
+  ) {}
+
+  // TODO: Check if the instrument scientist is the technical reviewer (or manager of instrument).
```
[the remark this answers is not in the document]
Comment (martin-trajanovski) [reply apps/user-office-backend/src/mutations/InternalReviewMutations.ts:21]: @enthusiastio I think the access control is defined and tested in the next PR coming after this one. Here is the link: https://github.com/UserOfficeProject/user-office-core/pull/259 but it should be reviewed right after this is merged.

**Pull request merged**